src/Funds/FundDetails/FundDetailsContainer.js

```diff
@@ -68,7 +75,16 @@ const FundDetailsContainer = ({
             path: `${LEDGERS_API}/${fundResponse.fund.ledgerId}/current-fiscal-year`,
           });
         })
-        .then(setCurrentFY)
+        .then(currentFYResponse => {
+          setCurrentFY(currentFYResponse);
+
+          return mutator.currentBudget.GET({
+            params: {
+              query: `fundId==${params.id} and fiscalYearId==${currentFYResponse.id}`,
+            },
+          });
+        })
+        .then((budgetResponse) => setCurrentBudget(budgetResponse[0]))
         .catch(() => setCompositeFund({ fund: {}, groupIds: [] }))
```

Comment:
this `catch` is related to the first `then`, but now it's not. Maybe it makes sense to move it inside corresponding `then` callback

Reply:
done